### PR TITLE
Add optional project filter to features generate types command

### PR DESCRIPTION
**Pull request by danielz-mnx**

### CODE REVIEW

**Comment by DeanVanNiekerk** [comment]
++ on this

**Comment by sohara** [comment]
👍 This is important for orgs codebases that organize flags by projects, where a project represents an application within the org.

**Comment by arudenkoofficial** [comment]
+1
Have the same issue for the  pro payment plan.
@tinahollygb Is it possible to expedite the approval of this PR somehow?

**Comment by artzhookov** [comment]
We have several projects. Features and experiments are loaded only for a specific project, but types are loaded for all.
It's a little annoying. Let's merge this pull request and fix the problem.

**Comment by august-growthbook** [comment]
Hi all, the person who was managing this repo is no longe at GrowthBook. I'll raise this internally since there have been some upvotes lately. Thanks for your time and interest.

File: src/utils/feature.ts [hunk 1]
```diff
@@ -29,9 +29,10 @@ export const getFeatureValueTypeToTypeScriptMapping = (valueType: string): strin
  * Fetches all the features from the GrowthBook REST API for the provided API base URL.
  * @param apiBaseUrl  The API base URL for the target GrowthBook instance
  * @param token  Secret token
+ * @param projectId  Optional project ID filter
  * @return {Promise<SimplifiedFeature[]>} A list of features
  */
-export const fetchAllPaginatedFeatures = async (apiBaseUrl: string, token: string): Promise<SimplifiedFeature[]> => {
+export const fetchAllPaginatedFeatures = async (apiBaseUrl: string, token: string, projectId?: string): Promise<SimplifiedFeature[]> => {
```
Comment (romain-growthbook) [inline src/utils/feature.ts:35]:
```suggestion
export const fetchAllPaginatedFeatures = async ({ apiBaseUrl, token,  projectId } : { apiBaseUrl: string, token: string, projectId?: string}): Promise<SimplifiedFeature[]> => {
```

Let's switch to using an object param for readibility and extensivity with future optional parameters.

**Review by romain-growthbook — COMMENTED** [review]
The PR looks good except for some suggested switch to a single argument param object.

**Comment by romain-growthbook** [comment]
I merged the PR. I'd like to refactor some of the code but this would have too much to carry for this one.

**Pull request merged**